pkg/modulewriter/modulewriter.go

```diff
@@ -373,14 +386,26 @@ func prepDepDir(depDir string, overwrite bool) error {
 	return nil
 }
 
+func prepArtifactsDir(deploymentio deploymentio.Deploymentio, artifactsDir string) error {
+	err := os.Mkdir(artifactsDir, 0700)
+	if err != nil && !os.IsExist(err) {
```

Comment:
I would change to MkdirAll so you are not dependent on creation of prevGroupDir which happens outside this function. I also would not allow the case where the dir already exists. We just deleted it so it should not be there.

Reply:
Due to the shuffling of "RemoveAll", I agree it makes sense not to ignore any non-nil err. I'm not entirely sold on MkdirAll but will add it.

Seeing this code with fresh eyes, I think the passing of `deploymentio.Deploymentio` is probably over-complicating what ought to be a simple file write and I've removed that in favor of simple `os` calls.